main.py

```diff
@@ -15,54 +17,67 @@
+            w_estimates = np.zeros(N)
+            phi_estimates = np.zeros(N)
 
-            w_estimates[k] = omega_hat
-            phi_estimates[k] = phi_hat
+            status_bar_progress = 0
+            run_time_begin = dt.now()
+            for k in range(N):
+                x_d = sig.x_discrete(SNR_dB)
 
-            status_bar_progress = utility.print_status_bar(k, status_bar_progress, N)
+                omega_hat, phi_hat, _, _ = fft_estimator.estimator(x_d, M)
 
-        mean_w = np.mean(w_estimates)
-        mean_phi = np.mean(phi_estimates)
+                w_estimates[k] = omega_hat
+                phi_estimates[k] = phi_hat
 
-        var_w = np.var(w_estimates)
-        var_phi = np.var(phi_estimates)
+                status_bar_progress = utility.print_status_bar(k, status_bar_progress, N)
 
-        crlb_w = crlb.omega(SNR_dB)
-        crlb_phi = crlb.phi(SNR_dB)
+            mean_w = np.mean(w_estimates) / m
+            mean_phi = np.mean(phi_estimates)
 
-        run_time_end = dt.now()
-        print("")
-        utility.print_execution_time(run_time_begin, run_time_end)
-        if var_w < crlb_w:
-            print("Variance for omega lower than CRLB!")
+            var_w = np.var(w_estimates)
+            var_phi = np.var(phi_estimates)
 
-        if var_phi < crlb_phi:
-            print("Variance for phi lower than CRLB!")
-            
+            crlb_w = crlb.omega(SNR_dB)
+            crlb_phi = crlb.phi(SNR_dB)
 
-        print("CONFIG | SNR [dB]: {}, M: 2^{}, true omega: {}, true phase: {}".format(SNR_dB, K, cfg.w0, cfg.phi))
-        print("OMEGA  | estimated mean: {}, estimated variance: {}, crlb: {}".format(mean_w, var_w, crlb_w))
-        print("PHASE  | estimated mean: {}, estimated variance: {}, crlb: {}".format(mean_phi, var_phi, crlb_phi))
-        print("")
+            run_time_end = dt.now()
+            print("")
+            utility.print_execution_time(run_time_begin, run_time_end)
 
-total_time_end = dt.now()
-utility.print_execution_time(total_time_begin, total_time_end)
+            w_estimate_valid = True
+            phi_estimate_valid = True
+            if var_w < crlb_w:
+                w_estimate_valid = False
+                print("Variance for omega lower than CRLB!")
+
+            if var_phi < crlb_phi:
+                phi_estimate_valid = False
+                print("Variance for phi lower than CRLB!")
+                
+
+            writer.writerow([SNR_dB, K, crlb_w, var_w, w_estimate_valid, crlb_phi, var_phi, phi_estimate_valid, mean_w, mean_phi])
+
+            # print("CONFIG | SNR [dB]: {}, M: 2^{}, true omega: {}, true phase: {}".format(SNR_dB, K, cfg.w0, cfg.phi))
+            # print("OMEGA  | estimated mean: {}, estimated variance: {}, crlb: {}".format(mean_w, var_w, crlb_w))
+            # print("PHASE  | estimated mean: {}, estimated variance: {}, crlb: {}".format(mean_phi, var_phi, crlb_phi))
+            # print("")
```

Comment:
Kommentere ut?

Reply:
Kommenterte ut for speed, men slenger de inn igjen

Reply:
I fix